### ISSUE TITLE
Problematic mappings for <a> without href

### ISSUE THREAD

**Issue by aleventhal** [w3c/html-aam#333]
An `<a name>` should not be mapped to text, because it has a text child.
This would lead to a a11y API structure where text, which should be a leaf, would not be a leaf and have a text child.

Rather, `<a name>` could be mapped to the ARIA "generic" role, which did not exist when HTML-AAM was first written.

**Comment by scottaohara** [comment]
hi @aleventhal.  yes, the `<a>` without `href` does need to be updated. mapping to `generic `makes sense, but for the additional work for 'inline' generic elements that still needs to be specified - https://github.com/w3c/aria/issues/960  (a w/out href not listed in that issue, but it would be included) - i know there are other issues/threads on this topic, but i can't find them right now.

**Comment by aleventhal** [comment]
@scottaohara are sure that w3c/aria#960 is a dependency? 
Really an ```<a name>``` is no different from ```<span id>```. Both can be used as an in-page link target. They don't really need to be exposed to the AT but if they are, it should be done just a generic/grouping.

**Comment by scottaohara** [comment]
unless thoughts have changed on that topic.  But, I've added the agenda label so we can talk about this soon (next week?) in the wg call

**Comment by jnurthen** [comment]
From Meeting on 2021/06/10
[Problematic mappings for <a> without href](https://github.com/w3c/html-aam/issues/333)
scott: comes back to "generic inline" etc.

jnurthen: it's not really the role that makes it generic inline or block.

scott: right. comes down to platform mappings. One currently says it maps to text.
… that's why I'd like to get consensus from the group

jnurthen: what does span map to>?

joanie: ATK maps it to generic.
… if my platform is the only blocker, I'm ok with changing it.

scott: a no href for IA1 maps role system text / text frame
… but we have different mappings, same with div

joanie: maybe we can check with NVDA people. IA2 are parallel to ATK mappings. So ask them if it's a problem.
… if they agree, then we can make progress.
… as a follow up conversation, make inline vs block generic a separate issue
… taking CSS into account
… I think we can take CSS into consideration for platform roles

scott: sounds good with me

aaron: do we agree that a without href can just map as span with id?

all: sounds good

<siri> agree as <span> as it is used a place holder when we <a> without href

aaron: speaking with James Teh, they use a different role for div vs span
… he agreed it was kind of nice but probably didn't matter.
… and we can tackle block vs inline (and other display) separately
